substrate/primitives/staking/src/lib.rs

```diff
@@ -100,19 +100,29 @@ pub trait OnStakingUpdate<AccountId, Balance> {
 	///
 	/// This is effectively any changes to the bond amount, such as bonding more funds, and
 	/// unbonding.
-	fn on_stake_update(_who: &AccountId, _prev_stake: Option<Stake<Balance>>) {}
+	fn on_stake_update(
+		_who: &AccountId,
+		_prev_stake: Option<Stake<Balance>>,
```

Comment:
Should we have both `_prev_stake` and `_stake` as same type? Either `Stake<Balance>` or `Option<Stake<Balance>>`. If `Stake<Balance>`, we can pass a zero value.